march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -212,6 +227,14 @@ void MarchHardwareInterface::read(ros::Duration elapsed_time)
     joint_velocity_[i] = filters::exponentialSmoothing(joint_velocity, joint_velocity_[i], 0.2);
 
     ROS_DEBUG("Joint %s: read position %f", joint_names_[i].c_str(), joint_position_[i]);
+
+    this->updateIMotionCubeState(i);
+  }
+
+  // Publish IMotionCubeStates if possible
+  if (realtime_pubs_->trylock())
```

Comment:
You only need to do `this->updateIMotionCubeState(i);` when you are actually planning on publishing